pkg/common/volume_operations.go

```diff
@@ -662,3 +670,21 @@ func (vo *VolumeOperationsImpl) fillCache() {
 		vo.cache.Set(volume.Name, volume.Namespace)
 	}
 }
+
+// getVolumeAppLabel returns App name related with PVC
+func (vo *VolumeOperationsImpl) getVolumeAppLabel(ctx context.Context, pvcName, pvcNamespace string) (string, error) {
+	ll := vo.log.WithFields(logrus.Fields{
+		"method": "getVolumeAppLabel",
+	})
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := vo.k8sClient.Get(ctx, client.ObjectKey{
+		Name:      pvcName,
+		Namespace: pvcNamespace,
+	}, pvc); err != nil {
+		ll.Errorf("Failed to get PVC %s in namespace %s, error %v", pvcName, pvcNamespace, err)
+		return "", err
+	}
+
+	return pvc.GetLabels()[k8s.AppLabelKey], nil
```

Comment:
How do we handle case when annotation is not set?

Reply:
We set empty label `app.kubernetes.io/name=""`

Reply:
fixed